stale caches -- any way to avoid or flush?

### Welcome

- [X] Yes, I understand that the GitHub action repository is not the repository of golangci-lint itself.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).

### Description of the problem

I maintain https://github.com/tmc/langchaingo and have encountered several apparent stale caches.

Locally and currently in CI I see

```
$ make lint
golangci-lint run --color=always --sort-results ./...
chains/llm.go:88:1: GetMemory returns interface (github.com/tmc/langchaingo/schema.Memory) (ireturn)
// GetOutputKeys returns the output keys the chain will return.
^
```

locally a cache clean fixes this but I don't see any way to do that with this action

### Version of golangci-lint

golangci-lint has version 1.52.2 built with go1.20.2 from da04413 on 2023-03-23T16:18:48Z

### Version of the GitHub Action

3.6.0

### Workflow file
```yml
name: ci

on:
  push:
  pull_request:
    branches:
      - main

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/setup-go@v4
        with:
          go-version: stable
          # Cache is managed by golangci-lint
          # https://github.com/actions/setup-go#caching-dependency-files-and-build-outputs
          cache: false
      - uses: actions/checkout@v3
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v3.6.0
  build-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/setup-go@v4
        with:
          go-version: stable
          cache: false
      - uses: actions/checkout@v3
      - name: Build
        run: go build -v ./...
      - name: Test
        run: go test -v ./...
```
### Go version

1.20.5

### Code example or link to a public repository

<details>

https://github.com/tmc/langchaingo/actions/runs/5363122543/jobs/9730477729

</details>

--- a/README.md
+++ b/README.md
@@ -195,6 +195,23 @@ with:
   # ...
 ```
 
+### `cache-invalidation-interval`
+
+(optional)
+
+Periodically invalidate the cache every `cache-invalidation-interval` days to ensure that outdated data is removed and fresh data is loaded.
+
+The default value is `7`.
+
+```yml
+uses: golangci/golangci-lint-action@v5
+with:
+  cache-invalidation-interval: 15
+  # ...
+```
+
+If set the number is `<= 0`, the cache will be always invalidate (Not recommended).
+
 ### `annotations`
 
 (optional)

--- a/action.yml
+++ b/action.yml
@@ -44,6 +44,10 @@ inputs:
     description: "golangci-lint command line arguments"
     default: ""
     required: false
+  cache-invalidation-interval:
+    description: "Periodically invalidate a cache because a new code being added. (number of days)"
+    default: '7'
+    required: false
 runs:
   using: "node20"
   main: "dist/run/index.js"

--- a/src/cache.ts
+++ b/src/cache.ts
@@ -25,6 +25,11 @@ const getLintCacheDir = (): string => {
 
 const getIntervalKey = (invalidationIntervalDays: number): string => {
   const now = new Date()
+
+  if (invalidationIntervalDays <= 0) {
+    return `${now.getTime()}`
+  }
+
   const secondsSinceEpoch = now.getTime() / 1000
   const intervalNumber = Math.floor(secondsSinceEpoch / (invalidationIntervalDays * 86400))
   return intervalNumber.toString()
@@ -32,21 +37,28 @@ const getIntervalKey = (invalidationIntervalDays: number): string => {
 
 async function buildCacheKeys(): Promise<string[]> {
   const keys = []
+
+  const invalidationIntervalDays = parseInt(core.getInput(`cache-invalidation-interval`, { required: true }).trim())
+
   // Periodically invalidate a cache because a new code being added.
-  // TODO: configure it via inputs.
-  let cacheKey = `golangci-lint.cache-${getIntervalKey(7)}-`
+  let cacheKey = `golangci-lint.cache-${getIntervalKey(invalidationIntervalDays)}-`
   keys.push(cacheKey)
+
   // Get working directory from input
   const workingDirectory = core.getInput(`working-directory`)
+
   // create path to go.mod prepending the workingDirectory if it exists
   const goModPath = path.join(workingDirectory, `go.mod`)
+
   core.info(`Checking for go.mod: ${goModPath}`)
+
   if (await pathExists(goModPath)) {
     // Add checksum to key to invalidate a cache when dependencies change.
     cacheKey += await checksumFile(`sha1`, goModPath)
   } else {
     cacheKey += `nogomod`
   }
+
   keys.push(cacheKey)
 
   return keys

